### PR TITLE
Fix docs (easy)

### PR DESCRIPTION
**Pull request by huitseeker**
Improves rustdoc formatting, fixes one error which is throwing up our CI:
https://github.com/lurk-lab/arecibo/actions/workflows/docs.yml

### CODE REVIEW

File: src/spartan/batched_ppsnark.rs [hunk 1]
```diff
@@ -1307,11 +1307,11 @@ impl<E: Engine, EE: EvaluationEngineTrait<E>> BatchedRelaxedR1CSSNARK<E, EE> {
     ))
   }
 
-  /// In round i, computes the evaluations at X_i = 0, 2, 3 of the univariate polynomials S(X_i)
+  /// In round i, computes the evaluations at `X_i` = 0, 2, 3 of the univariate polynomials `S(X_i`)
```
Comment (mpenciak) [inline src/spartan/batched_ppsnark.rs:1310]:
```suggestion
  /// In round i, computes the evaluations at `X_i` = 0, 2, 3 of the univariate polynomials `S(X_i)`
```

File: src/spartan/batched_ppsnark.rs [hunk 2]
```diff
@@ -1116,9 +1116,9 @@ impl<E: Engine, EE: EvaluationEngineTrait<E>> BatchedRelaxedR1CSSNARK<E, EE> {
   /// Note that at the end of the protocol, the prover returns the evaluation
   ///   u' = P'(r_{0},...,r_{n-1}) = P(r_{n-m},...,r_{n-1})
   /// However, the polynomial we actually committed to over {0,1}^n is
-  ///   P''(X_{0},...,X_{n-1}) = L_0(X_{0},...,X_{n-m-1}) * P(X_{n-m},...,X_{n-1})
+  ///   P''(X_{0},...,X_{n-1}) = `L_0(X`_{0},...,X_{n-m-1}) * P(X_{n-m},...,X_{n-1})
```
Comment (mpenciak) [inline src/spartan/batched_ppsnark.rs:1119]: I don't know if this is right?

File: src/spartan/batched_ppsnark.rs [hunk 3]
```diff
@@ -1116,9 +1116,9 @@ impl<E: Engine, EE: EvaluationEngineTrait<E>> BatchedRelaxedR1CSSNARK<E, EE> {
...
   /// The SNARK prover/verifier will need to rescale the evaluation by the first Lagrange polynomial
-  ///   u'' = L_0(r_{0},...,r_{n-m-1}) * u'
+  ///   u'' = `L_0(r`_{0},...,r_{n-m-1}) * u'
```
Comment (mpenciak) [inline src/spartan/batched_ppsnark.rs:1121]: Same here

File: src/provider/hyperkzg.rs [hunk 4]
```diff
@@ -1,5 +1,5 @@
 //! This module implements Nova's evaluation engine using `HyperKZG`, a KZG-based polynomial commitment for multilinear polynomials
-//! HyperKZG is based on the transformation from univariate PCS to multilinear PCS in the Gemini paper (section 2.4.2 in https://eprint.iacr.org/2022/420.pdf).
+//! HyperKZG is based on the transformation from univariate PCS to multilinear PCS in the Gemini paper (section 2.4.2 in `<https://eprint.iacr.org/2022/420.pdf>`).
```
Comment (mpenciak) [inline src/provider/hyperkzg.rs:2]: Is this the correct way to denote URLs in markdown?

Comment (huitseeker) [reply src/provider/hyperkzg.rs:2]: https://doc.rust-lang.org/beta/rustdoc/lints.html#bare_urls

**Review by mpenciak — APPROVED** [review]

**Pull request merged**